src/image.rs

```diff
@@ -143,6 +144,41 @@ pub enum ImageFormat {
     HDR,
 }
 
+impl ImageFormat {
+    /// Return the image format specified by the path's file extension.
+    pub fn from_path<P>(path: P) -> ImageResult<Self> where P : AsRef<Path> {
```

Comment:
In my other rework I had, internally, something similar as `from_path<P>(path: P) -> Option<ImageFormat>`. It's not truely an error here, though I'm not sure which version fits better and `ImageResult<Self>` is the most generic option. On the one hand, there are file extensions which do not specify an image but on the other hand we do not support all image formats.

Reply:
Usually I consider Result to be a better API than Option, and it's easy to convert to an option with ok().

Here a hypothetical case for result, although it's a little much and at the end of the day you're the maintainer and I will make the API to be what you're willing to support.

The nice thing is Result gives context over *why* the result is not ok and be able to special case certain failures, for example, a user may in the future want to special case when the extension isn't specified and fall back to guessing the format from the magic bytes. Currently the error message for the missing extension case isn't great.

Reply:
Wanting to differentiate various causes and having a better error message is a good point, so let's keep it as a `ImageResult<_>` with a note to #928/#921 which should eventually provide better distinction of the cases.